phpBB/phpbb/notification/type/mention.php

```diff
@@ -0,0 +1,404 @@
+		$this->set_data('post_subject', $post['post_subject']);
+		$this->set_data('post_username', (($post['poster_id'] == ANONYMOUS) ? $post['post_username'] : ''));
+		$this->set_data('forum_id', $post['forum_id']);
+		$this->set_data('forum_name', $post['forum_name']);
+		$this->notification_time = $post['post_time'];
+		parent::create_insert_array($post, $pre_create_data);
+	}
+
+	/**
+	* Get the HTML formatted title of this notification
+	*
+	* @return string
+	*/
+	public function get_title()
+	{
+		$responders = $this->get_data('responders');
+		$usernames = array();
+
+		if (!is_array($responders))
+		{
+			$responders = array();
+		}
+
+		$responders = array_merge(array(array(
+			'poster_id'     => $this->get_data('poster_id'),
+			'username'      => $this->get_data('post_username'),
+		)), $responders);
+
+		$responders_cnt = count($responders);
+		$responders = $this->trim_user_ary($responders);
+		$trimmed_responders_cnt = $responders_cnt - count($responders);
+
+		foreach ($responders as $responder)
+		{
+			if ($responder['username'])
+			{
+				$usernames[] = $responder['username'];
+			}
+			else
+			{
+				$usernames[] = $this->user_loader->get_username($responder['poster_id'], 'no_profile');
+			}
+		}
+
+		if ($trimmed_responders_cnt > 20)
+		{
+			$usernames[] = $this->language->lang('NOTIFICATION_MANY_OTHERS');
+		}
+		else if ($trimmed_responders_cnt)
+		{
+			$usernames[] = $this->language->lang('NOTIFICATION_X_OTHERS', $trimmed_responders_cnt);
+		}
+
+		return $this->language->lang(
+			$this->language_key,
+			phpbb_generate_string_list($usernames, $this->user),
+			$responders_cnt
+		);
+	}
+
+	/**
+	* Users needed to query before this notification can be displayed
+	*
+	* @return array Array of user_ids
+	*/
+	public function users_to_query()
+	{
+		$responders = $this->get_data('responders');
+		$users = array(
+			$this->get_data('poster_id'),
+		);
+
+		if (is_array($responders))
+		{
+			foreach ($responders as $responder)
+			{
+				$users[] = $responder['poster_id'];
+			}
+		}
+
+		return $this->trim_user_ary($users);
+	}
+
+	/**
+	* Trim the user array passed down to 3 users if the array contains
+	* more than 4 users.
+	*
+	* @param array $users Array of users
+	* @return array Trimmed array of user_ids
+	*/
+	public function trim_user_ary($users)
+	{
+		if (count($users) > 4)
+		{
+			array_splice($users, 3);
+		}
+		return $users;
+	}
+
+	/**
+	* Get the HTML formatted reference of the notification
+	*
+	* @return string
+	*/
+	public function get_reference()
+	{
+		return $this->language->lang(
+			'NOTIFICATION_REFERENCE',
+			censor_text($this->get_data('topic_title'))
+		);
+	}
+
+	/**
+	* Add responders to the notification
+	*
+	* @param mixed $post
+	* @return array Array of responder data
+	*/
+	public function add_responders($post)
+	{
+		// Do not add them as a responder if they were the original poster that created the notification
+		if ($this->get_data('poster_id') == $post['poster_id'])
```

Comment:
@paul999  I was actually thinking of getting rid of some of the methods that we don't need for now in mention.php. I did this initially because base class was being inherited but now that post class is inherited I don't think these extra methods are necessary until and unless we don't have a use for them for mention class.